Content.Server/DropPod/DropPodSystem.cs

```diff
@@ -0,0 +1,342 @@
+using Content.Server.AlertLevel;
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Chat.Systems;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.Coordinates;
+using Content.Shared.DropPod;
+using Content.Shared.Maps;
+using Content.Shared.Weather;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using System.Threading;
+
+namespace Content.Server.DropPod
+{
+    public sealed class DropPodSystem : EntitySystem
+    {
+        [Dependency] private readonly AtmosphereSystem _atmosphere = default!;
+        [Dependency] private readonly AlertLevelSystem _alertLevelSystem = default!;
+        [Dependency] private readonly ChatSystem _chat = default!;
+        [Dependency] private readonly EntityLookupSystem _lookupSystem = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager = default!;
+        [Dependency] private readonly SharedMapSystem _mapSystem = default!;
+        [Dependency] private readonly SharedTransformSystem _transform = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<DropPodConsoleComponent, BoundUIOpenedEvent>(OnBoundUIOpened);
+            SubscribeLocalEvent<DropPodConsoleComponent, DropPodRefreshMessage>(OnRefreshLandingButtonPressed);
+            SubscribeLocalEvent<DropPodConsoleComponent, DropPodStartMessage>(OnStartLandingButtonPressed);
+            SubscribeLocalEvent<DropPodConsoleComponent, DropPodPointSelectedMessage>(OnPointSelected);
+        }
+
+        private int UIN = 0;
+        private bool canActivate = false;
+        private float top_border = 0f;
+        private float bottom_border = 0f;
+        private float left_border = 0f;
+        private float right_border = 0f;
+
+        #region Ui
+        private void OnPointSelected(EntityUid uid, DropPodConsoleComponent component, DropPodPointSelectedMessage args)
+        {
+            UIN = args.Point; // we remember the unique identification number of the point, for further work with it
+            canActivate = true; // we give you the opportunity to start disembarking
+
+            UpdateUI(uid);
+        }
+
+        private void OnBoundUIOpened(EntityUid uid, DropPodConsoleComponent component, BoundUIOpenedEvent args)
+        {
+            canActivate = false;
+
+            UpdateUI(uid);
+        }
+
+
+        private void OnRefreshLandingButtonPressed(EntityUid uid, DropPodConsoleComponent component, DropPodRefreshMessage message)
+        {
+            canActivate = false;
+
+            UpdateUI(uid);
+        }
+
+        private void UpdateUI(EntityUid uid)
+        {
```

Comment:
All UI updates and interactions should be in Shared, for better prediction and responsiveness